src/mappings/trade_reversions.ts

```diff
@@ -0,0 +1,47 @@
+import { log, BigInt, store } from '@graphprotocol/graph-ts'
+import { TradeReversion as TradeReversionEvent } from '../../generated/BatchExchange/BatchExchange'
+import { Batch, Solution, Trade } from '../../generated/schema'
+import { toEventId, toPriceId, getBatchId } from '../utils'
+import { createSolution } from './solution'
+import { updateOrderOnTradeReversion } from './orders'
+
+export function onTradeReversion(event: TradeReversionEvent): void {
+  let batchId = getBatchId(event).minus(BigInt.fromI32(1))
+  let batch = Batch.load(batchId.toString())!
+  let solution = Solution.load(batch.solution)!
```

Comment:
I would recommend using the `get${entity_name}ById` pattern that we use for other entities. This allows you to assert that the entity is not-null with an `if` statement. AFAIK, `!` just tells the compiler to assume that it is not null and will potentially produce a "null reference error" later.